src/mg/PAMI/Client/Impl/ClientImpl.php

```diff
@@ -405,20 +405,21 @@ public function send(OutgoingMessage $message)
 	    if (@fwrite($this->_socket, $messageToSend) < $length) {
     	    throw new ClientException('Could not send message');
 	    }
-	    $read = 0;
-	    while($read <= $this->_rTimeout) {
-	        $this->process();
```

Comment:
currently the while(1) loop is taking care of that. I was not sure about your definition of rTimeout and cTimeout.
My interpretation was/is:
rTimeout = time to wait for any action response / if we get anything then wait again for tTimeout time.
cTimeout = fixed time to wait for events to come in, from the moment of starting (this is currently not fully handled i believe).